pkg/relay/processmanager.go

```diff
@@ -0,0 +1,178 @@
+package relay
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/blocknative/dreamboat/pkg/structs"
+	"github.com/flashbots/go-boost-utils/bls"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	ResponseQueueSubmit = iota
+	ResponseQueueRegister
+	ResponseQueueOther
+)
+
+type ProcessManager struct {
+	LastRegTime map[string]uint64 // [pubkey]timestamp
+	lrtl        sync.RWMutex      // LastRegTime RWLock
+
+	VerifySubmitBlockCh       chan VerifyReq
+	VerifyRegisterValidatorCh chan VerifyReq
+	VerifyOtherCh             chan VerifyReq
+
+	StoreCh chan StoreReq
+
+	m ProcessManagerMetrics
+}
+
+func NewProcessManager(verifySize, storeSize uint) *ProcessManager {
+	rm := &ProcessManager{
+		LastRegTime: make(map[string]uint64),
+
+		VerifySubmitBlockCh:       make(chan VerifyReq, verifySize),
+		VerifyRegisterValidatorCh: make(chan VerifyReq, verifySize),
+		VerifyOtherCh:             make(chan VerifyReq, verifySize),
+
+		StoreCh: make(chan StoreReq, storeSize),
+	}
+	rm.initMetrics()
+	return rm
+}
+
+func (rm *ProcessManager) RunVerify(num uint) {
+	for i := uint(0); i < num; i++ {
+		go rm.VerifyParallel()
+	}
+}
+
+func (rm *ProcessManager) RunStore(store Datastore, ttl time.Duration, num uint) {
+	for i := uint(0); i < num; i++ {
+		go rm.ParallelStore(store, ttl)
+	}
+}
+
+func (rm *ProcessManager) RunCleanup(checkinterval uint64, cleanupInterval time.Duration) {
+	for {
+		now := uint64(time.Now().Unix())
+		var keys []string
+		rm.lrtl.RLock()
+		for k, v := range rm.LastRegTime {
+			if checkinterval < now-v {
+				keys = append(keys, k)
+			}
+		}
+		rm.lrtl.RUnlock()
+
+		rm.lrtl.Lock()
+		for _, k := range keys {
```

Comment:
(This is the second for loop in question.  Creating a comment so that we can explicitly resolve.)

Reply:
done